session_test.go

```diff
@@ -405,19 +405,17 @@ func TestNegotiateStreamError(t *testing.T) {
 	defer cancel()
 	clientConn, serverConn := net.Pipe()
 	clientJID := jid.MustParse("me@example.net")
+	semaphore := make(chan struct{})
 	go func() {
-		s, err := xmpp.ReceiveSession(ctx, serverConn, 0, xmpp.NewNegotiator(xmpp.StreamConfig{
+		_, err := xmpp.ReceiveSession(ctx, serverConn, 0, xmpp.NewNegotiator(xmpp.StreamConfig{
```

Comment:
Out of curiosity, why did you remove closing the session? I don't know why it was there in the first place as the session should error before reaching it, but I don't think it affects the race condition whether it's there or not and I'd like to keep the diff as minimal as possible and not have extraneous changes. Unless I'm missing something and it changes things?

Reply:
oops, nevermind, forgot this was in your original post